sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSource.scala

```diff
@@ -116,12 +116,16 @@ case class StateSourceOptions(
     batchId: Long,
     operatorId: Int,
     storeName: String,
-    joinSide: JoinSideValues) {
+    joinSide: JoinSideValues,
+    snapshotStartBatchId: Option[Long],
+    snapshotPartitionId: Option[Int]) {
   def stateCheckpointLocation: Path = new Path(resolvedCpLocation, DIR_NAME_STATE)
 
   override def toString: String = {
     s"StateSourceOptions(checkpointLocation=$resolvedCpLocation, batchId=$batchId, " +
-      s"operatorId=$operatorId, storeName=$storeName, joinSide=$joinSide)"
+      s"operatorId=$operatorId, storeName=$storeName, joinSide=$joinSide" +
+      snapshotStartBatchId.map(", snapshotStartBatchId=" + _).getOrElse("") +
```

Comment:
what does this look like when the options are not present ? maybe just do a simple `if else` condition otherwise ?

Reply:
This line is equivalent to 
```
{
  if (snapshotStartBatchId.isDefined) {
    ", snapshotStartBatchId=" + snapshotStartBatchId.get
  } else {
    ""
  }
}
```
In my opinion, it is a little less elegant.

Reply:
I'd be OK to print them as None if they aren't provided - in case this could make thing easier.

If we want to apply some logic to create a part of string, I'd prefer having them as local variables. It doesn't sound to be very intuitive to mix up string interpolation and string appends with logic to create a string.